pages/programdetails/[pid].tsx

```diff
@@ -1,153 +1,110 @@
-                teacherName: "Brian",
-            },
-            {
-                image: "https://image.flaticon.com/icons/png/512/2395/2395826.png",
-                name: "Singing Giraffes",
-                ageGroup: "9 & Under",
-                spaceAvailable: 5,
-                volunteerSpaceAvailable: 4,
-                weekday: weekday.THU,
-                startTimeMinutes: 1110,
-                durationMinutes: 60,
-                teacherName: "Brian",
-            },
-        ],
-        "1": [
-            {
-                image: "https://media.giphy.com/media/jEyKIvmt0BgLC/giphy.gif",
-                name: "Class 1",
-                ageGroup: "10 & Under",
-                spaceAvailable: 2,
-                volunteerSpaceAvailable: 4,
-                weekday: weekday.MON,
-                startTimeMinutes: 1234,
-                durationMinutes: 60,
-                teacherName: "Teacher1",
-            },
-            {
-                image: "https://i.pinimg.com/originals/fb/f7/64/fbf76458d74b5236cb7ee9208563bebc.gif",
-                name: "Class 2",
-                ageGroup: "10 & Under",
-                spaceAvailable: 1,
-                volunteerSpaceAvailable: 4,
-                weekday: weekday.THU,
-                startTimeMinutes: 765,
-                durationMinutes: 60,
-                teacherName: "Teacher2",
-            },
-        ],
-    };
-    // TODO this will be a ProgramCardInfo (once the reverted PR gets unreverted)
-    const programInfo: { [programId: string]: any } = {
-        "0": {
-            name: "Building Bridges with Music",
-            description:
-                "Children with special needs will be able to connect with the music teacher through an online video call to socialize and have fun while learning about music!",
-            image: "",
-            startDate: "July 7, 2021",
-            endDate: "August 28, 2021",
-            tag: "Music",
-            format: "Online",
-        },
-        "1": {
-            name: "Education Through Creativity",
-            description:
-                "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
-            image: "",
-            startDate: "July 7, 2021",
-            endDate: "August 28, 2021",
-            tag: "Art",
-            format: "Online",
-        },
-    };
 
-    return programName in programInfo ? (
+    const fetchWithId = (url, id) =>
+        fetch(`${url}?id=${id}`).then((r) => r.json());
+    const { data: classListResponse, error: classListError } = useSWR(
+        ["/api/cardinfo/class", pid],
+        fetchWithId,
+    );
+    const { data: programInfoResponse, error: programInfoError } = useSWR(
+        ["/api/cardinfo/program", pid],
+        fetchWithId,
+    );
+    if (classListError || programInfoError)
+        return (
+            <Text>
+                An error has occurred.{" "}
```

Comment:
what's the `.{" "}` for?

Reply:
LOL i have no idea, maybe i left an extra space somewhere and the formatter did this

Reply:
we can remove it then